src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ApplicationPropertyPage.xaml

```diff
@@ -104,6 +104,42 @@
     </StringProperty.ValueEditors>
   </StringProperty>
 
+  <!-- TODO use fwlink for HelpUrl -->
+  <BoolProperty Name="UseWindowsForms"
+                DisplayName="Windows Forms"
+                Description="Enable Windows Forms for this project."
+                HelpUrl="https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props-desktop"
+                Category="General">
+    <BoolProperty.Metadata>
+      <NameValuePair Name="DependsOn" Value="Application::OutputType" />
+      <NameValuePair Name="VisibilityCondition">
+        <NameValuePair.Value>
+          (and
+            (has-evaluated-value "Application" "OutputType" "WinExe")
```

Comment:
You probably want to force `WinExe` over `Exe` if either of these frameworks are selected: https://review.docs.microsoft.com/help/contribute/links-how-to?branch=master#locale-codes-in-external-links

Reply:
Is that the right link? I'm not sure what you mean by forcing here.

Reply:
Of course not 🤦‍♂️ Here's the right link: https://docs.microsoft.com/dotnet/core/compatibility/sdk/5.0/automatically-infer-winexe-output-type#change-description

![image](https://user-images.githubusercontent.com/4403806/119951463-53d4cf80-bfdf-11eb-88af-ff7920c897e8.png)
This what prompted the comment about forcing - e.g. is a user checked `UseWindowsForms` or `UseWPF` we shouldn't allow switching to `Exe`. May be even remove it from the list?
Likewise may be show a checkbox for `DisableWinExeOutputInference`?

Reply:
Interesting. Offhand I'm not sure we can dynamically refresh the contents of the output type list.

What we _could_ do is have the `UseWindowsForms` and `UseWPF` properties be deleted from the project if the user selects an output type other than `WinExe`.

Reply:
Strictly speaking Exe is an accepted value for a Windows Forms app, it is the same as WinExe as of .NET 5.0.
Also one can build a DLL targeting Windows Forms, e.g. GitUI or GitCommands are both libraries targeting Windows Forms.

Reply:
The Property Pages UI will reflect the final evaluated value of the OutputType property. So for WinForms or WPF apps if you try to set it to Exe in the UI the project file will get updated--and then the UI will reset itself because the value is overridden in the targets.

This means the `DependsOn` is backwards--the OutputType effectively depends on `UseWindowsForms` and `UseWPF` even though for the purposes of showing these controls we would like it to be the other way around.

I agree with @drewnoakes; let's get this in and see what the feedback is. It's better to have these checkboxes in some form now rather than wait until we've figured out the perfect implementation.